test_xml2_rpm_2_9_8 is failing

In #125 I added some new file tests and test_xml2_rpm_2_9_8 is failing.

test_xml2_rpm_2_9_1 is passing, so things aren't totally wrong.  test_xml2_rpm_2_9_1 is from centos and test_xml2_rpm_2_9_8 is from fedora, so I would have expected them to be pretty similar.

```
/tmp/cve-bin-tool-cki5f550/libxml2-2.9.8-4.fc29.x86_64.rpm.extracted/usr/lib64/libxml2.so.2.9.8 is xml2 
Known CVEs in version 
CVE-2015-8806, CVE-2016-4483, CVE-2016-4614, CVE-2016-4615, CVE-2016-4616, CVE-2016-9596, CVE-2016-9598, CVE-2017-15412, CVE-2017-18258, CVE-2017-5130, CVE-2017-7376
F
======================================================================
FAIL: test_xml2_rpm_2_9_8 (test.test_scanner.TestScanner)
Test detection on libxml2 2.9.8 fedora rpm
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/terri/Code/cve-bin-tool/test/test_scanner.py", line 580, in test_xml2_rpm_2_9_8
    '2.9.8')
  File "/home/terri/Code/cve-bin-tool/test/test_scanner.py", line 110, in _file_test
    self.assertIn(version, cves[package])
AssertionError: '2.9.8' not found in {'': {'CVE-2015-8806': CVE(number='CVE-2015-8806', version='-', severity='HIGH'), 'CVE-2016-4483': CVE(number='CVE-2016-4483', version='-', severity='HIGH'), 'CVE-2016-4614': CVE(number='CVE-2016-4614', version='-', severity='CRITICAL'), 'CVE-2016-4615': CVE(number='CVE-2016-4615', version='-', severity='CRITICAL'), 'CVE-2016-4616': CVE(number='CVE-2016-4616', version='-', severity='CRITICAL'), 'CVE-2016-9596': CVE(number='CVE-2016-9596', version='-', severity='MEDIUM'), 'CVE-2016-9598': CVE(number='CVE-2016-9598', version='-', severity='MEDIUM'), 'CVE-2017-15412': CVE(number='CVE-2017-15412', version='-, 1.7.0, 1.7.1, 1.7.2, 1.7.3, 1.7.4, 1.8.0, 1.8.1, 1.8.2, 1.8.3, 1.8.4, 1.8.5, 1.8.6, 1.8.7, 1.8.9, 1.8.10, 1.8.13, 1.8.14, 1.8.16, 2.0.0, 2.1.0, 2.1.1, 2.2.0, 2.2.1, 2.2.2, 2.2.3, 2.2.4, 2.2.5, 2.2.6, 2.2.7, 2.2.8, 2.2.9, 2.2.10, 2.2.11, 2.3.0, 2.3.1, 2.3.2, 2.3.3, 2.3.4, 2.3.5, 2.3.6, 2.3.7, 2.3.8, 2.3.9, 2.3.10, 2.3.11, 2.3.12, 2.3.13, 2.3.14, 2.4.1, 2.4.2, 2.4.3, 2.4.4, 2.4.5, 2.4.6, 2.4.7, 2.4.8, 2.4.9, 2.4.10, 2.4.11, 2.4.12, 2.4.13, 2.4.14, 2.4.15, 2.4.16, 2.4.17, 2.4.18, 2.4.19, 2.4.20, 2.4.21, 2.4.22, 2.4.23, 2.4.24, 2.4.25, 2.4.26, 2.4.27, 2.4.28, 2.4.29, 2.4.30, 2.5.0, 2.5.4, 2.5.7, 2.5.8, 2.5.10, 2.5.11, 2.6.0, 2.6.1, 2.6.2, 2.6.3, 2.6.4, 2.6.5, 2.6.6, 2.6.7, 2.6.8, 2.6.9, 2.6.11, 2.6.12, 2.6.13, 2.6.14, 2.6.16, 2.6.17, 2.6.18, 2.6.20, 2.6.21, 2.6.22, 2.6.23, 2.6.24, 2.6.25, 2.6.26, 2.6.27, 2.6.28, 2.6.29, 2.6.30, 2.6.31, 2.6.32, 2.7.0, 2.7.1, 2.7.2, 2.7.3, 2.7.4, 2.7.5, 2.7.6, 2.7.7, 2.7.8, 2.8.0, 2.9.0, 2.9.1, 2.9.2, 2.9.3, 2.9.4', severity='HIGH'), 'CVE-2017-18258': CVE(number='CVE-2017-18258', version='-, 1.7.0, 1.7.1, 1.7.2, 1.7.3, 1.7.4, 1.8.0, 1.8.1, 1.8.2, 1.8.3, 1.8.4, 1.8.5, 1.8.6, 1.8.7, 1.8.9, 1.8.10, 1.8.13, 1.8.14, 1.8.16, 2.0.0, 2.1.0, 2.1.1, 2.2.0, 2.2.1, 2.2.2, 2.2.3, 2.2.4, 2.2.5, 2.2.6, 2.2.7, 2.2.8, 2.2.9, 2.2.10, 2.2.11, 2.3.0, 2.3.1, 2.3.2, 2.3.3, 2.3.4, 2.3.5, 2.3.6, 2.3.7, 2.3.8, 2.3.9, 2.3.10, 2.3.11, 2.3.12, 2.3.13, 2.3.14, 2.4.1, 2.4.2, 2.4.3, 2.4.4, 2.4.5, 2.4.6, 2.4.7, 2.4.8, 2.4.9, 2.4.10, 2.4.11, 2.4.12, 2.4.13, 2.4.14, 2.4.15, 2.4.16, 2.4.17, 2.4.18, 2.4.19, 2.4.20, 2.4.21, 2.4.22, 2.4.23, 2.4.24, 2.4.25, 2.4.26, 2.4.27, 2.4.28, 2.4.29, 2.4.30, 2.5.0, 2.5.4, 2.5.7, 2.5.8, 2.5.10, 2.5.11, 2.6.0, 2.6.1, 2.6.2, 2.6.3, 2.6.4, 2.6.5, 2.6.6, 2.6.7, 2.6.8, 2.6.9, 2.6.11, 2.6.12, 2.6.13, 2.6.14, 2.6.16, 2.6.17, 2.6.18, 2.6.20, 2.6.21, 2.6.22, 2.6.23, 2.6.24, 2.6.25, 2.6.26, 2.6.27, 2.6.28, 2.6.29, 2.6.30, 2.6.31, 2.6.32, 2.7.0, 2.7.1, 2.7.2, 2.7.3, 2.7.4, 2.7.5, 2.7.6, 2.7.7, 2.7.8, 2.8.0, 2.9.0, 2.9.1, 2.9.2, 2.9.3, 2.9.4, 2.9.5', severity='MEDIUM'), 'CVE-2017-5130': CVE(number='CVE-2017-5130', version='-, 1.7.0, 1.7.1, 1.7.2, 1.7.3, 1.7.4, 1.8.0, 1.8.1, 1.8.2, 1.8.3, 1.8.4, 1.8.5, 1.8.6, 1.8.7, 1.8.9, 1.8.10, 1.8.13, 1.8.14, 1.8.16, 2.0.0, 2.1.0, 2.1.1, 2.2.0, 2.2.1, 2.2.2, 2.2.3, 2.2.4, 2.2.5, 2.2.6, 2.2.7, 2.2.8, 2.2.9, 2.2.10, 2.2.11, 2.3.0, 2.3.1, 2.3.2, 2.3.3, 2.3.4, 2.3.5, 2.3.6, 2.3.7, 2.3.8, 2.3.9, 2.3.10, 2.3.11, 2.3.12, 2.3.13, 2.3.14, 2.4.1, 2.4.2, 2.4.3, 2.4.4, 2.4.5, 2.4.6, 2.4.7, 2.4.8, 2.4.9, 2.4.10, 2.4.11, 2.4.12, 2.4.13, 2.4.14, 2.4.15, 2.4.16, 2.4.17, 2.4.18, 2.4.19, 2.4.20, 2.4.21, 2.4.22, 2.4.23, 2.4.24, 2.4.25, 2.4.26, 2.4.27, 2.4.28, 2.4.29, 2.4.30, 2.5.0, 2.5.4, 2.5.7, 2.5.8, 2.5.10, 2.5.11, 2.6.0, 2.6.1, 2.6.2, 2.6.3, 2.6.4, 2.6.5, 2.6.6, 2.6.7, 2.6.8, 2.6.9, 2.6.11, 2.6.12, 2.6.13, 2.6.14, 2.6.16, 2.6.17, 2.6.18, 2.6.20, 2.6.21, 2.6.22, 2.6.23, 2.6.24, 2.6.25, 2.6.26, 2.6.27, 2.6.28, 2.6.29, 2.6.30, 2.6.31, 2.6.32, 2.7.0, 2.7.1, 2.7.2, 2.7.3, 2.7.4, 2.7.5, 2.7.6, 2.7.7, 2.7.8, 2.8.0, 2.9.0, 2.9.1, 2.9.2, 2.9.3, 2.9.4', severity='HIGH'), 'CVE-2017-7376': CVE(number='CVE-2017-7376', version='-', severity='CRITICAL')}}

----------------------------------------------------------------------
```

--- a/cve_bin_tool/checkers/xml2.py
+++ b/cve_bin_tool/checkers/xml2.py
@@ -17,6 +17,8 @@ def guess_xml2_version_from_content(lines):
     new_guess = ""
     pattern1 = re.compile(r"\/libxml2\-([0-9]+\.[0-9]+\.[0-9]+)\/")
     pattern2 = re.compile(r"\\libxml2\-([0-9]+\.[0-9]+\.[0-9]+)\\")
+    # fedora 29 string looks like libxml2.so.2.9.8-2.9.8-4.fc29.x86_64.debug
+    pattern3 = re.compile(r"libxml2.so.([0-9]+\.[0-9]+\.[0-9]+)")
 
     for line in lines:
         match = pattern1.search(line)
@@ -38,6 +40,12 @@ def guess_xml2_version_from_content(lines):
             new_guess = "2.9.3"
         if line == "20904":
             new_guess = "2.9.4"
+
+        match = pattern3.search(line)
+        if match:
+            new_guess2 = match.group(1).strip()
+            if len(new_guess2) > len(new_guess):
+                new_guess = new_guess2
     return new_guess
 
 

